editor/plugins/animation_library_editor.cpp

```diff
@@ -0,0 +1,689 @@
+			undo_redo->add_do_method(al.ptr(), "add_animation", anim_name, animd);
+			undo_redo->add_undo_method(al.ptr(), "remove_animation", anim_name);
+			undo_redo->add_undo_method(al.ptr(), "add_animation", anim_name, anim);
+			undo_redo->add_do_method(this, "_update_editor", player);
+			undo_redo->add_undo_method(this, "_update_editor", player);
+			undo_redo->commit_action();
+		} break;
+		case FILE_MENU_EDIT_ANIMATION: {
+			EditorNode::get_singleton()->push_item(anim.ptr());
+		} break;
+	}
+}
+void AnimationLibraryEditor::_load_file(String p_path) {
+	switch (file_dialog_action) {
+		case FILE_DIALOG_ACTION_OPEN_LIBRARY: {
+			Ref<AnimationLibrary> al = ResourceLoader::load(p_path);
+			if (al.is_null()) {
+				error_dialog->set_text(TTR("Invalid AnimationLibrary file"));
+				error_dialog->popup_centered();
+				return;
+			}
+
+			TypedArray<StringName> libs = player->call("get_animation_library_list");
+			for (int i = 0; i < libs.size(); i++) {
+				const StringName K = libs[i];
+				Ref<AnimationLibrary> al2 = player->call("get_animation_library", K);
+				if (al2 == al) {
+					error_dialog->set_text(TTR("This library is already added to the player."));
+					error_dialog->popup_centered();
+
+					return;
+				}
+			}
+
+			String name = p_path.get_file().get_basename();
+
+			int attempt = 1;
+
+			while (bool(player->call("has_animation_library", name))) {
+				attempt++;
+				name = p_path.get_file().get_basename() + " " + itos(attempt);
+			}
+
+			UndoRedo *undo_redo = EditorNode::get_singleton()->get_undo_redo();
+
+			undo_redo->create_action(vformat(TTR("Add Animation Library: %s"), name));
+			undo_redo->add_do_method(player, "add_animation_library", name, al);
+			undo_redo->add_undo_method(player, "remove_animation_library", name);
+			undo_redo->add_do_method(this, "_update_editor", player);
+			undo_redo->add_undo_method(this, "_update_editor", player);
+			undo_redo->commit_action();
+		} break;
+		case FILE_DIALOG_ACTION_OPEN_ANIMATION: {
+			Ref<Animation> anim = ResourceLoader::load(p_path);
+			if (anim.is_null()) {
+				error_dialog->set_text(TTR("Invalid Animation file"));
+				error_dialog->popup_centered();
+				return;
+			}
+
+			Ref<AnimationLibrary> al = player->call("get_animation_library", adding_animation_to_library);
+			List<StringName> anims;
+			al->get_animation_list(&anims);
+			for (const StringName &K : anims) {
+				Ref<Animation> a2 = al->get_animation(K);
+				if (a2 == anim) {
+					error_dialog->set_text(TTR("This animation is already added to the library."));
+					error_dialog->popup_centered();
+					return;
+				}
+			}
+
+			String name = p_path.get_file().get_basename();
+
+			int attempt = 1;
+
+			while (al->has_animation(name)) {
+				attempt++;
+				name = p_path.get_file().get_basename() + " " + itos(attempt);
+			}
+
+			UndoRedo *undo_redo = EditorNode::get_singleton()->get_undo_redo();
+
+			undo_redo->create_action(vformat(TTR("Load Animation into library: %s"), name));
```

Comment:
We seem to use Title Case for action names.
```suggestion
			undo_redo->create_action(vformat(TTR("Load Animation into Library: %s"), name));
```